jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java

```diff
@@ -650,30 +660,18 @@ public HttpCookie getSessionCookie(HttpSession session, String contextPath, bool
             sessionPath = (StringUtil.isEmpty(sessionPath)) ? "/" : sessionPath;
             String id = getExtendedId(session);
             HttpCookie cookie = null;
-            if (_sessionComment == null)
-            {
-                cookie = new HttpCookie(
-                    _cookieConfig.getName(),
-                    id,
-                    _cookieConfig.getDomain(),
-                    sessionPath,
-                    _cookieConfig.getMaxAge(),
-                    _cookieConfig.isHttpOnly(),
-                    _cookieConfig.isSecure() || (isSecureRequestOnly() && requestIsSecure));
-            }
-            else
-            {
-                cookie = new HttpCookie(
-                    _cookieConfig.getName(),
-                    id,
-                    _cookieConfig.getDomain(),
-                    sessionPath,
-                    _cookieConfig.getMaxAge(),
-                    _cookieConfig.isHttpOnly(),
-                    _cookieConfig.isSecure() || (isSecureRequestOnly() && requestIsSecure),
-                    _sessionComment,
-                    1);
-            }
+
+            cookie = new HttpCookie(
+                _cookieConfig.getName(),
+                id,
+                _cookieConfig.getDomain(),
+                sessionPath,
+                _cookieConfig.getMaxAge(),
+                _cookieConfig.isHttpOnly(),
+                _cookieConfig.isSecure() || (isSecureRequestOnly() && requestIsSecure),
+                HttpCookie.getCommentWithoutAttributes(_cookieConfig.getComment()),
+                1,
```

Comment:
Is this `1` for version 1 cookies?
That's deprecated.

Reply:
This was a question. :nerd_face: 
Didn't mean for you to blindly change it to 0.

Reply:
The old code used "1" if there was a Session Comment, and didn't use "1" (not sure what default value is) if there wasn't a Session Comment.

https://github.com/eclipse/jetty.project/blob/e09444eeb52c3454c4b78cd27ec47187fc44700e/jetty-server/src/main/java/org/eclipse/jetty/server/session/SessionHandler.java#L653-L676

Reply:
I think the whole usage of versioned cookies is deprecated, so I don't think we should every use version 1, even if we have comments.

Reply:
Versioned cookies are used.
Old installations still use it.
And the Servlet TCK mandates it.

Reply:
correct TCK test use version 1. I will run a tck test with the branch to see what happen.

Reply:
fyi doesn't break any cookie tck tests (https://jenkins.webtide.net/job/nightlies/job/servlettck-run-jetty-9.4.x/323/) please note tck tests are run using 
```
jetty.httpConfig.requestCookieCompliance=RFC2965
jetty.httpConfig.responseCookieCompliance=RFC2965
```